images/nginx/healthcheck/healthz.locations

```diff
@@ -0,0 +1,8 @@
+location /.lagoonhealthz {
+ content_by_lua_block {
+  ngx.status = ngx.HTTP_OK;
+  ngx.header.content_type = 'application/json';
+  ngx.say('{"check_nginx":"pass"}');
+  ngx.exit(ngx.OK);
+ }
+}
```

Comment:
is this file supposed to be here?

Reply:
Hi @Schnitzel - yeah, I was half and half about including it - it's essentially a duplicate of the Lua one - but I was thinking through edge cases and I was worried that there _may_ be a case where the conditional that copies the "disabled" files doesn't run, but the reference to the `healthz.locationz` file might still be there:

I was wondering what might happen if the entrypoint was overwritten, but since the helper files are there, and the healthz.locations file is referenced _from inside a helper_, you might get the situation where there was a reference to a file that isn't there (since the entrypoint file no longer copies `healthz.locations`)

But then I got really paranoid and though, okay, that's fine, maybe I ship the default as LUA and only have the entrypoint copy the php version across if it's set ... but then I thought "if I do that, I'll be overwriting the LUA default and won't be able to get it back until the container is recreated).

So I went with the, admittedly, non-DRY solution on this - I'm happy to reign in my paranoia though.
The options are:

1 - have the duplication, but be assured of always having a healthz.location file existing (maybe it could be an empty file with some comments in it? I haven't tried this yet)

2 - have the default be Lua (so we're guaranteed a healthz.locations file exists), and only ever overwrite with the php version

3 - don't ship a default healthz.locations file, and accept that if someone (or some script) doesn't explicitly copy over one of the .disabled files, nginx is gonna die.

Which do you think is best?

Reply:
@bomoko 
Thanks for outlining your thought process! Yes I agree all versions 1-3 would make sense. I think though that the current state is confusing and looks like a bug, so I'm worried that people that want to understand what happens here are having a hard time to understand it.
therefore I think version 2 is the best, it actually fits to what we already do in the nginx image:
we ship with a static file version for `app.config` in the `nginx` image: https://github.com/amazeeio/lagoon/blob/93490ded5be6ac97f5415995c9d577f917d6ec69/images/nginx/Dockerfile#L37
which is then overwritten by the nginx-drupal: https://github.com/amazeeio/lagoon/blob/93490ded5be6ac97f5415995c9d577f917d6ec69/images/nginx-drupal/Dockerfile#L10

so I think the system of overwriting is a good one.

Reply:
Great! Done. The build failed, so I'll kick off a new one - tested, all seems to be back in order.